### PR TITLE
Fix OVN NetworkManager subclass spec failures

### PR DESCRIPTION
**Pull request by agrare**
Fixes https://travis-ci.com/github/ManageIQ/manageiq-providers-ovirt/jobs/503285272#L2196-L2356

Depends:
- [x] https://github.com/ManageIQ/manageiq-providers-openstack/pull/714

### CODE REVIEW

**Comment by agrare** [comment]
Cross-repo tests are green: https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/364

**Comment by chessbyte** [comment]
is oVirt really using classes from Openstack?

**Comment by agrare** [comment]
Yes, the OVN network manager is essentially an Openstack compatible neutron API so we re-use the openstack collector/parser/etc...

https://www.ovirt.org/develop/release-management/features/network/ovirt-ovn-provider.html
https://www.ovirt.org/develop/release-management/features/network/manageiq_ovn.html

File: app/models/manageiq/providers/redhat/network_manager/cloud_network/private.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ManageIQ::Providers::Redhat::NetworkManager::CloudNetwork::Private < ManageIQ::Providers::Redhat::NetworkManager::CloudNetwork
+  def self.display_name(number = 1)
+    n_('Cloud Network (Ovift)', 'Cloud Networks (Ovirt)', number)
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/network_manager/cloud_network/private.rb:3]:
```suggestion
    n_('Cloud Network (Ovirt)', 'Cloud Networks (Ovirt)', number)
```

File: app/models/manageiq/providers/redhat/network_manager/cloud_network/public.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork::Public < ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/network_manager/cloud_network/public.rb:1]:
```suggestion
class ManageIQ::Providers::Redhat::NetworkManager::CloudNetwork::Public < ManageIQ::Providers::Openstack::NetworkManager::CloudNetwork
```

File: app/models/manageiq/providers/redhat/network_manager/cloud_network/private.rb (hunk 1 above)
Comment (agrare) [reply app/models/manageiq/providers/redhat/network_manager/cloud_network/private.rb]: :facepalm: fixed

**Comment by Fryguy** [comment]
@agrare Does this require a data migration?  It's hard to tell, but it looks like Public and Private are new classes.

**Comment by agrare** [comment]
Actually those are existing Openstack classes but in order to have them show up as subclasses of `ManageIQ::Providers::Redhat::NetworkManager::CloudNetwork` they had to inherit from that rather than their Openstack equivalents.

And yes we need a data migration to update the type for any Redhat::NetworkManager

**Comment by Fryguy** [comment]
`morphy` version: https://github.com/ManageIQ/manageiq-providers-ovirt/pull/569

**Pull request merged**